packages/core/src/__integrationtests__/PublicCredentials.spec.ts

```diff
@@ -0,0 +1,315 @@
+      credentialId,
+      encodedAssetCredential
+    )
+    let encodedAllAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const allAssetCredentialsBeforeRevocation =
+      await PublicCredential.credentialsFromChain(encodedAllAssetCredentials)
+    // Verify that credential was revoked before un-revocation
+    expect(assetCredential.revoked).toBe(true)
+
+    const unrevocationTx = api.tx.publicCredentials.unrevoke(credentialId, null)
+    const authorizedTx = await Did.authorizeTx(
+      attester.uri,
+      unrevocationTx,
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(authorizedTx, tokenHolder)
+
+    encodedAssetCredential = await api.call.publicCredentials.getCredential(
+      credentialId
+    )
+    assetCredential = await PublicCredential.credentialFromChain(
+      credentialId,
+      encodedAssetCredential
+    )
+    encodedAllAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const allAssetCredentialsAfterRevocation =
+      await PublicCredential.credentialsFromChain(encodedAllAssetCredentials)
+
+    // Verify it is now not revoked anymore
+    expect(assetCredential.revoked).toBe(false)
+    // Verify the number of credentials has not changed after revocation
+    expect(allAssetCredentialsBeforeRevocation.length).toEqual(
+      allAssetCredentialsAfterRevocation.length
+    )
+  })
+
+  it('should be possible to remove a credential', async () => {
+    const credentialId = PublicCredential.computeId(
+      latestCredential,
+      attester.uri
+    )
+    let encodedAssetCredential = await api.call.publicCredentials.getCredential(
+      credentialId
+    )
+    let encodedAllAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const allAssetCredentialsBeforeRevocation =
+      await PublicCredential.credentialsFromChain(encodedAllAssetCredentials)
+    // Verify that credential existed before removal
+    expect(encodedAssetCredential.isNone).toBe(false)
+
+    const removalTx = api.tx.publicCredentials.remove(credentialId, null)
+    const authorizedTx = await Did.authorizeTx(
+      attester.uri,
+      removalTx,
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(authorizedTx, tokenHolder)
+
+    encodedAssetCredential = await api.call.publicCredentials.getCredential(
+      credentialId
+    )
+    encodedAllAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const allAssetCredentialsAfterRevocation =
+      await PublicCredential.credentialsFromChain(encodedAllAssetCredentials)
+
+    // Verify it is now removed
+    expect(encodedAssetCredential.isNone).toBe(true)
+    // Verify the number of credentials has decreased by one
+    expect(allAssetCredentialsBeforeRevocation.length).toEqual(
+      allAssetCredentialsAfterRevocation.length + 1
```

Comment:
I was confused by mismatching order and "decreased" vs "+1". Could this be aligned?